sympy/functions/special/delta_functions.py

```diff
@@ -67,6 +67,8 @@ def eval(cls, arg, k=0):
             return S.NaN
         if arg.is_positive or arg.is_negative:
             return S.Zero
+        if fuzzy_not(im(arg).is_zero):
+            raise ValueError("Function defined only for Real Values. Complex part: %s  found in %s ." % (repr(im(arg)), repr(arg)) )
```

Comment:
I see this was already merged. Please follow a format like this in the future. You'll be able to do multiline messages easier.